azurerm/resource_arm_eventgrid_domain.go

```diff
@@ -0,0 +1,380 @@
+package azurerm
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/Azure/azure-sdk-for-go/services/preview/eventgrid/mgmt/2018-09-15-preview/eventgrid"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/response"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmEventGridDomain() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmEventGridDomainCreateUpdate,
+		Read:   resourceArmEventGridDomainRead,
+		Update: resourceArmEventGridDomainCreateUpdate,
+		Delete: resourceArmEventGridDomainDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"location": locationSchema(),
+
+			"resource_group_name": resourceGroupNameSchema(),
+
+			"tags": tagsSchema(),
+
+			"input_schema": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  string(eventgrid.InputSchemaEventGridSchema),
+				ForceNew: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					string(eventgrid.InputSchemaCloudEventV01Schema),
+					string(eventgrid.InputSchemaCustomEventSchema),
+					string(eventgrid.InputSchemaEventGridSchema),
+				}, true),
```

Comment:
could we make this case-sensitive? 
```suggestion
				}, false),
```